### ISSUE TITLE
Feed: wrong href for atom:link tag

### ISSUE THREAD

**Issue by Globulopolis** [joomla/joomla-cms#39775]
### Steps to reproduce the issue
Open any feed on site(e.g. featured)

### Expected result
Correct url in href
```html
<atom:link rel="self" type="application/rss+xml"
href="http://localhost/j42/index.php?option=com_content&amp;view=featured&amp;Itemid=101&amp;format=feed&amp;type=rss"/>
```

### Actual result
Double `j42/j42`
```html
<atom:link rel="self" type="application/rss+xml"
href="http://localhost/j42/j42/index.php?option=com_content&amp;view=featured&amp;Itemid=101&amp;format=feed&amp;type=rss"/>
```

### System information (as much as possible)
Apache 2.4, PHP 7.2.
No cache, no gzip.

### Additional comments
The problem is coming when Joomla output the feed. If I do `echo $url . $syndicationURL;` right after https://github.com/joomla/joomla-cms/blob/4.2-dev/libraries/src/Document/Renderer/Feed/RssRenderer.php#L93 I see the correct url.

Same problem with the atom feed.

**Comment by brianteeman** [comment]
Confirmed

**Comment by Hackwar** [comment]
I only have that issue when SEF is switched off.

**Comment by brianteeman** [comment]
@hackwar iirc the key here is that the site has to be installed in a subdirectory

**Comment by Hackwar** [comment]
I have it in a subdirectory and it is working fine with SEF enabled. It indeed fails when SEF is disabled.

**Comment by brianteeman** [comment]
retested and I confirm @Hackwar findings

I would say its still a bug although very low priority

**Comment by EJBJane** [comment]
Again, as I mentioned in another issue about the RSS this is a big problem for SEO. You can see errors in the RSS in my feed here. As Globulopolis said he got double directories /j42/j42/ - I get this also but WITH SEF on, links in articles which were added with the link manager and in images urls. As I cant turn the RSS feed off I get lot's of 404's because of the large amount of traffic I have. 
See feed with full article here: https://www.dolcevia.com/en/articles?format=feed&type=rss<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39775">issues.joomla.org/tracker/joomla-cms/39775</a>.</sub>

**Comment by Hackwar** [comment]
I created a PR with #42802 Please test this, so that we can fix it in 5.0 and onwards. Closing this since we have a PR.
